engine/src/main/java/org/terasology/world/propagation/LocalChunkView.java

```diff
@@ -2,29 +2,25 @@
 // SPDX-License-Identifier: Apache-2.0
 package org.terasology.world.propagation;
 
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-import org.terasology.math.ChunkMath;
-import org.terasology.math.geom.Vector3i;
+import org.joml.Vector3ic;
 import org.terasology.world.block.Block;
 import org.terasology.world.chunks.Chunk;
-import org.terasology.world.chunks.ChunkConstants;
+import org.terasology.world.chunks.Chunks;
 
 /**
  * Provides a simple view over some chunks using a propagation rule.
  */
 public class LocalChunkView implements PropagatorWorldView {
-    private static final Logger logger = LoggerFactory.getLogger(LocalChunkView.class);
 
     private PropagationRules rules;
     private Chunk[] chunks;
 
-    private final Vector3i topLeft;
+    private final org.joml.Vector3i topLeft;
 
     public LocalChunkView(Chunk[] chunks, PropagationRules rules) {
         this.chunks = chunks;
         this.rules = rules;
-        topLeft = chunks[0].getPosition();
+        topLeft = chunks[0].getPosition(new org.joml.Vector3i());
```

Comment:
how many instances of joml and math.geom vectors do we have here? Does it maybe make sense to import joml and explicitly mention math.geom vectors instead of the other way around?

Reply:
I didn't check, but we'll need to do another pass over this package anyways.

I'm going to continue with the module PRs, and then remove the deprecated API in here. Together with further migration of chunk concepts we'll get rid of TeraMath eventually...